chain/indexer/src/streamer/mod.rs

```diff
@@ -106,24 +106,25 @@ pub async fn build_streamer_message(
         let views::ChunkView { transactions, author, header, receipts: chunk_non_local_receipts } =
             chunk;
 
-        let mut outcomes = shards_outcomes
+        let outcomes = shards_outcomes
             .remove(&header.shard_id)
             .expect("Execution outcomes for given shard should be present");
-
-        // Take execution outcomes for receipts from the vec and keep only the ones for transactions
-        let mut receipt_outcomes = outcomes.split_off(transactions.len());
-
+        let outcome_count = outcomes.len();
+        let mut outcomes = outcomes
+            .into_iter()
+            .map(|outcome| (outcome.execution_outcome.id, outcome))
+            .collect::<BTreeMap<_, _>>();
+        debug_assert_eq!(outcomes.len(), outcome_count);
         let indexer_transactions = transactions
             .into_iter()
-            .zip(outcomes.into_iter())
-            .map(|(transaction, outcome)| {
-                assert_eq!(
-                    outcome.execution_outcome.id, transaction.hash,
-                    "This ExecutionOutcome must have the same id as Transaction hash"
-                );
-                IndexerTransactionWithOutcome { outcome, transaction }
+            .filter_map(|transaction| {
+                let outcome = outcomes.remove(&transaction.hash)?;
```

Comment:
We're missing the guarantee that the outcome was present
```suggestion
                let outcome = outcomes.remove(&transaction.hash).expect("Missing ExecutionOutcome for a transaction");
```

Reply:
That's because it isn't required for all the transactions to produce an outcome in our current protocol.

Reply:
@evgenykuzyakov, thanks for your input here! This is a library used by other services as well. We want to avoid crashing and allow users to define the behaviour themselves.